### PR TITLE
bump MbedTLS to 3.6.0

### PR DESCRIPTION
**Pull request by tomi-font**
https://github.com/Mbed-TLS/mbedtls/releases/tag/v3.6.0

I reapplied the patches applied in https://github.com/zephyrproject-rtos/mbedtls/pull/52 after merging, but
- TF-M patches (https://github.com/TrustedFirmware-M/trusted-firmware-m/tree/main/lib/ext/mbedcrypto) 0001, 0004 and 0006 conflicted. I resolved the conflicts as I thought made sense, effectively keeping the same intended changes, but some of the patches may need to be updated differently. @adeaarm
- all the patches other than the TF-M ones have been addressed upstream and so aren't needed here anymore.

### CODE REVIEW

**Comment by adeaarm** [comment]
I will shortly make patches 0001 to 0006 on top of Mbed TLS 3.6.0 available on review.trustedfirmware.org so you can directly cherry-pick those. I will post here when the change is ready to be picked up.

**Review by frkv — APPROVED** [review]
LGTM

File: CMakeLists.txt [hunk 1]
```diff
@@ -283,11 +283,6 @@ if(LIB_INSTALL_DIR)
     set(CMAKE_INSTALL_LIBDIR "${LIB_INSTALL_DIR}")
 endif()
 
-if (NOT EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/framework/CMakeLists.txt")
-    message(FATAL_ERROR "${CMAKE_CURRENT_SOURCE_DIR}/framework/CMakeLists.txt not found. Run `git submodule update --init` from the source tree to fetch the submodule contents.")
-endif()
-add_subdirectory(framework)
-
```
Comment (mswarowsky) [inline CMakeLists.txt]: To make our future life during updates easier you could add to the commit message that this commit reverts some of the work of the commits:
469f7811fac98617e544aac2b1b2db56040c95db 
and 
f9bbe0de4c97ca532d052106bb471756e55742ae

Or even just add revert commits of these to commits

**Review by mswarowsky — APPROVED** [review]
One small NIT but lese lgtm

**Review by ceolin — CHANGES_REQUESTED** [review]
@tomi-font It seems good. Can you sign off the your commits ? They are misssing

**Review by ceolin — APPROVED** [review]

**Comment by ceolin** [comment]
Thanks @tomi-font !

**Review by d3zd3z — APPROVED** [review]
Presumably, this is just the reverts, the merge, and necessary changes.

**Pull request merged**